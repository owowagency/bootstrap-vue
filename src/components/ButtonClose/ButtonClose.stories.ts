import ButtonClose from '.';

export default {
    title: 'Components/Button close',
    component: ButtonClose,
};

const template = (args) => ({
    components: {ButtonClose},
    setup: () => ({args}),
    template: '<ButtonClose v-bind="args" />',
});

export const Default = template.bind({});

export const Disabled = template.bind({});

Disabled.args = {
    disabled: true,
};

