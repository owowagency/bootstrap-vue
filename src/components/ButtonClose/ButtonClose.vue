<template>
    <button
        class="btn-close"
        :disabled="disabled"
    />
</template>

<script lang="ts">
import {disabledProps} from '@/composables/useDisabled';
</script>

<script lang="ts" setup>
defineProps({
    ...disabledProps,
});
</script>

<docs>
```vue
<ButtonClose />
```

## Imported Props

| Prop name | Description            | Type    | Values | Default |
| --------- | ---------------------- | ------- | ------ | ------- |
| disabled  | Disables the component | boolean | -      | false   |
</docs>
