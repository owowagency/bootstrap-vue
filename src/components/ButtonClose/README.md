# ButtonClose

```vue
<ButtonClose />
```

## Imported Props

| Prop name | Description            | Type    | Values | Default |
| --------- | ---------------------- | ------- | ------ | ------- |
| disabled  | Disables the component | boolean | -      | false   |
