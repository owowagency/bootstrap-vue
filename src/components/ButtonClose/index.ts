import ButtonClose from './ButtonClose.vue';

export default ButtonClose;
