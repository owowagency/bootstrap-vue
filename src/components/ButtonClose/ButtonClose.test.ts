import {describe, expect, it} from 'vitest';
import ButtonClose from '.';
import {shallowMount} from '@vue/test-utils';

describe('template', () => {
    componentRenderTest(ButtonClose);

    it('adds disabled property', () => {
        const wrapper = shallowMount(ButtonClose, {
            props: {disabled: true},
        });

        expect((wrapper.element as HTMLButtonElement).disabled).toBe(true);
    });
});
