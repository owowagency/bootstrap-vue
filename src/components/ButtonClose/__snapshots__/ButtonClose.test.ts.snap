// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `"<button class=\\"btn-close\\"></button>"`;
